Exception when starting WebStorm after plugin installation

Then, a lot of [File Cache conflicts](http://www.jetbrains.com/webstorm/webhelp/file-cache-conflict.html) modal windows appears, and the only solution is to disactivate the plugin.
- Mac OS X 10.6.8
- JetBrain WebStorm 5.0.4

```
Error during dispatching of java.awt.event.WindowEvent[WINDOW_GAINED_FOCUS,opposite=com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog[dialog1,1054,671,866x139,invalid,hidden,layout=java.awt.BorderLayout,APPLICATION_MODAL,title=File Cache Conflict,defaultCloseOperation=DO_NOTHING_ON_CLOSE,rootPane=com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog$DialogRootPane[,0,22,866x117,invalid,layout=javax.swing.JRootPane$RootLayout,alignmentX=0.0,alignmentY=0.0,border=,flags=481,maximumSize=,minimumSize=,preferredSize=],rootPaneCheckingEnabled=true],oldState=0,newState=0] on frame1: Already disposed: Project (Disposed) vin
java.lang.AssertionError: Already disposed: Project (Disposed) vin
    at com.intellij.openapi.components.impl.ComponentManagerImpl.getComponent(ComponentManagerImpl.java:196)
    at com.intellij.openapi.fileEditor.FileEditorManager.getInstance(FileEditorManager.java:34)
    at org.editorconfig.configmanagement.CodeStyleManager.windowGainedFocus(CodeStyleManager.java:61)
    at java.awt.AWTEventMulticaster.windowGainedFocus(AWTEventMulticaster.java:406)
    at java.awt.Window.processWindowFocusEvent(Window.java:1918)
    at java.awt.Window.processEvent(Window.java:1833)
    at java.awt.Component.dispatchEventImpl(Component.java:4744)
    at java.awt.Container.dispatchEventImpl(Container.java:2141)
    at java.awt.Window.dispatchEventImpl(Window.java:2489)
    at java.awt.Component.dispatchEvent(Component.java:4572)
    at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1856)
    at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:920)
    at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:381)
    at java.awt.Component.dispatchEventImpl(Component.java:4616)
    at java.awt.Container.dispatchEventImpl(Container.java:2141)
    at java.awt.Window.dispatchEventImpl(Window.java:2489)
    at java.awt.Component.dispatchEvent(Component.java:4572)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:710)
    at java.awt.EventQueue.access$400(EventQueue.java:82)
    at java.awt.EventQueue$2.run(EventQueue.java:669)
    at java.awt.EventQueue$2.run(EventQueue.java:667)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:98)
    at java.awt.EventQueue$3.run(EventQueue.java:683)
    at java.awt.EventQueue$3.run(EventQueue.java:681)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:680)
    at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:718)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:543)
    at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:421)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:379)
    at java.awt.SequencedEvent.dispatch(SequencedEvent.java:101)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:708)
    at java.awt.EventQueue.access$400(EventQueue.java:82)
    at java.awt.EventQueue$2.run(EventQueue.java:669)
    at java.awt.EventQueue$2.run(EventQueue.java:667)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:98)
    at java.awt.EventQueue$3.run(EventQueue.java:683)
    at java.awt.EventQueue$3.run(EventQueue.java:681)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:680)
    at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:718)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:543)
    at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:421)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:379)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:296)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:211)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:196)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:188)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
```

--- a/src/org/editorconfig/plugincomponents/ConfigProjectComponent.java
+++ b/src/org/editorconfig/plugincomponents/ConfigProjectComponent.java
@@ -1,11 +1,9 @@
 package org.editorconfig.plugincomponents;
 
 import com.intellij.AppTopics;
-import com.intellij.openapi.Disposable;
 import com.intellij.openapi.components.ProjectComponent;
 import com.intellij.openapi.fileEditor.FileEditorManagerListener;
 import com.intellij.openapi.project.Project;
-import com.intellij.openapi.util.Disposer;
 import com.intellij.openapi.wm.IdeFrame;
 import com.intellij.openapi.wm.WindowManager;
 import com.intellij.util.messages.MessageBus;
@@ -48,15 +46,12 @@ public class ConfigProjectComponent implements ProjectComponent {
         IdeFrame frame = WindowManager.getInstance().getIdeFrame(project);
         final Window window = (Window)frame;
         window.addWindowFocusListener(codeStyleManager);
-        Disposer.register(project, new Disposable() {
-            @Override
-            public void dispose() {
-                window.removeWindowFocusListener(codeStyleManager);
-            }
-        });
     }
 
     public void projectClosed() {
         // called when project is being closed
+        IdeFrame frame = WindowManager.getInstance().getIdeFrame(project);
+        final Window window = (Window) frame;
+        window.removeWindowFocusListener(codeStyleManager);
     }
 }

--- a/src/org/editorconfig/plugincomponents/ConfigProjectComponent.java
+++ b/src/org/editorconfig/plugincomponents/ConfigProjectComponent.java
@@ -1,9 +1,11 @@
 package org.editorconfig.plugincomponents;
 
 import com.intellij.AppTopics;
+import com.intellij.openapi.Disposable;
 import com.intellij.openapi.components.ProjectComponent;
 import com.intellij.openapi.fileEditor.FileEditorManagerListener;
 import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Disposer;
 import com.intellij.openapi.wm.IdeFrame;
 import com.intellij.openapi.wm.WindowManager;
 import com.intellij.util.messages.MessageBus;
@@ -44,9 +46,14 @@ public class ConfigProjectComponent implements ProjectComponent {
     public void projectOpened() {
         // called when project is opened
         IdeFrame frame = WindowManager.getInstance().getIdeFrame(project);
-        Window window = (Window)frame;
+        final Window window = (Window)frame;
         window.addWindowFocusListener(codeStyleManager);
-
+        Disposer.register(project, new Disposable() {
+            @Override
+            public void dispose() {
+                window.removeWindowFocusListener(codeStyleManager);
+            }
+        });
     }
 
     public void projectClosed() {

